#! /bin/sh -e
## 21_newlist_help.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: make newlist be a bit friendlier.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
diff -urNad mailman-2.1.6~/bin/newlist mailman-2.1.6/bin/newlist
--- mailman-2.1.6~/bin/newlist	2005-12-10 13:37:50.000000000 +0100
+++ mailman-2.1.6/bin/newlist	2005-12-25 15:55:10.056996722 +0100
@@ -87,6 +87,9 @@
 defined in your Defaults.py file or overridden by settings in mm_cfg.py).
 
 Note that listnames are forced to lowercase.
+
+The list admin address need to be a fully-qualified address, like
+owner@example.com, not just owner.
 """
 
 import sys
@@ -94,6 +97,7 @@
 import getpass
 import getopt
 import sha
+import grp
 
 import paths
 from Mailman import mm_cfg
@@ -122,6 +126,9 @@
 
 
 def main():
+    gid = grp.getgrnam(mm_cfg.MAILMAN_GROUP)[2]
+    if os.getgid() != gid:
+        os.setgid(gid)
     try:
         opts, args = getopt.getopt(sys.argv[1:], 'hql:u:e:',
                                    ['help', 'quiet', 'language=',
@@ -199,7 +206,7 @@
         except Errors.BadListNameError, s:
             usage(1, _('Illegal list name: %(s)s'))
         except Errors.EmailAddressError, s:
-            usage(1, _('Bad owner email address: %(s)s'))
+            usage(1, _('Bad owner email address: %(s)s.  Owner addresses need to be fully-qualified names, like "owner@example.com", not just "owner".'))
         except Errors.MMListAlreadyExistsError:
             usage(1, _('List already exists: %(listname)s'))
 
